typespec-extension/src/code-model-builder.ts

```diff
@@ -548,121 +515,196 @@ export class CodeModelBuilder {
         // at present, use global security definition
         security: this.codeModel.security,
       });
-      codeModelClient.crossLanguageDefinitionId = client.crossLanguageDefinitionId;
+      // codeModelClient.crossLanguageDefinitionId = client.crossLanguageDefinitionId;
 
       // versioning
-      const versioning = getVersion(this.program, client.service);
-      if (versioning && versioning.getVersions()) {
-        // @versioned in versioning
+      const versions = client.apiVersions;
+      if (versions && versions.length > 0) {
         if (!this.sdkContext.apiVersion || ["all", "latest"].includes(this.sdkContext.apiVersion)) {
-          this.apiVersion = getDefaultApiVersion(this.sdkContext, client.service);
+          this.apiVersion = versions[versions.length - 1];
         } else {
-          this.apiVersion = versioning.getVersions().find((it: Version) => it.value === this.sdkContext.apiVersion);
+          this.apiVersion = versions.find((it: string) => it === this.sdkContext.apiVersion);
           if (!this.apiVersion) {
             throw new Error("Unrecognized api-version: " + this.sdkContext.apiVersion);
           }
         }
```

Comment:
Should we always take `versions[versions.length - 1]` ? As now TCGC handles the version.

@XiaofeiCao for opinion as well.

Reply:
Guess we can. For pinned scenario, seems TCGC now will filter out all api-versions that's larger than the pinned one. So the latest one will always be the pinned.

Reply:
TCGC also horners the sequence of the versions, if there is no pinned api version, the last one should be pinned API version. https://github.com/Azure/typespec-azure/pull/1045#discussion_r1678426224